lib/rubygems/package/io_source.rb

```diff
@@ -32,10 +32,14 @@ def present?
 
   def with_read_io
     yield io
+  ensure
+    io.rewind
   end
 
   def with_write_io
     yield io
+  ensure
+    io.rewind
```

Comment:
As far I understand, rewinding after write is not related to this PR, and you just added it for consistency? Seems fine, just wanted to confirm.